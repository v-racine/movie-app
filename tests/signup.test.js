const { Config } = require('../src/config/config');

Config.Get(process.env);

const AppFactory = require('../src/app');
const request = require('supertest');

describe('sign up', () => {
  const mockPgClient = {};

  const app = AppFactory({
    pgClient: mockPgClient,
  });

  describe('when: a user wants to view the form to signup', () => {
    let rsp;

    beforeEach(async () => {
      rsp = await request(app).get('/signUp').send();
    });

    afterEach(() => {
      jest.clearAllMocks();
    });

    test('then: we return the view of the form', async () => {
      const text = rsp.text;
      expect(text).toMatchSnapshot();

      const status = rsp.status;
      expect(status).toBe(200);
    });
  });

  describe("when a user attempts submits the 'sign up' form BUT the email already exists in the db", () => {});

  describe("when a user attempts submits the 'sign up' form BUT the password and password confirmation do NOT match", () => {});

  describe("when a user successfully submits the 'sign up' form", () => {});
});
