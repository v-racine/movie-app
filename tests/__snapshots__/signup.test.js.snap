// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`sign up when: a user wants to view the form to signup then: we return the view of the form 1`] = `
<!DOCTYPE html>
<html lang="en-US">
  <head>
    <title>
      The Ultimate Movie App
    </title>
    <meta charset="UTF-8">
    <link rel="stylesheet"
          href="/stylesheets/application.css"
    >
  </head>
  <body>
    <header>
      <h1>
        The Ultimate Movie App
      </h1>
    </header>
    <main>
      <form action="/signUp"
            method="post"
      >
        <dl>
          <dt>
            <label for="username">
              Enter a username:
            </label>
          </dt>
          <dd>
            <input type="text"
                   id="username"
                   name="username"
                   placeholder="name"
            >
          </dd>
          <dt>
            <label for="email">
              Enter a valid email:
            </label>
          </dt>
          <dd>
            <input type="email"
                   id="email"
                   name="email"
                   placeholder="username@example.com"
            >
          </dd>
          <dt>
            <label for="password">
              Create a password:
            </label>
          </dt>
          <dd>
            <input type="password"
                   id="password"
                   name="password"
            >
          </dd>
          <dt>
            <label for="passwordConfirmation">
              Confirm your password:
            </label>
          </dt>
          <dd>
            <input type="text"
                   id="passwordConfirmation"
                   name="passwordConfirmation"
            >
          </dd>
        </dl>
        <fieldset class="actions">
          <input type="submit"
                 value="Submit"
          >
          <input type="reset"
                 value="Clear Form"
          >
        </fieldset>
        <p>
        </p>
        <a href="/signin">
          Have an account? Sign In
        </a>
      </form>
    </main>
  </body>
</html>
`;
